controllers/controller/vpnkey_rotation_controller_test.go

```diff
@@ -283,16 +397,203 @@ var _ = Describe("VpnKeyRotation Controller", Ordered, func() {
+				Name:      cluster1.Name,
+			}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getKey, cluster1)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			cluster1.Status.CniSubnet = []string{"192.168.0.0/24"}
+			cluster1.Status.RegistrationStatus = v1alpha1.RegistrationStatusRegistered
+			Expect(k8sClient.Status().Update(ctx, cluster1)).Should(Succeed())
+
+			Expect(k8sClient.Create(ctx, cluster2)).Should(Succeed())
+			// update cluster status
+			getKey = types.NamespacedName{
+				Namespace: cluster2.Namespace,
+				Name:      cluster2.Name,
+			}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getKey, cluster2)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			cluster2.Status.CniSubnet = []string{"192.168.1.0/24"}
+			cluster2.Status.RegistrationStatus = v1alpha1.RegistrationStatusRegistered
+			Expect(k8sClient.Status().Update(ctx, cluster2)).Should(Succeed())
+
+			Expect(k8sClient.Create(ctx, cluster3)).Should(Succeed())
+			// update cluster status
+			getKey = types.NamespacedName{
+				Namespace: cluster3.Namespace,
+				Name:      cluster3.Name,
+			}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getKey, cluster3)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			cluster3.Status.CniSubnet = []string{"10.1.1.1/16"}
+			cluster3.Status.RegistrationStatus = v1alpha1.RegistrationStatusRegistered
+			Expect(k8sClient.Status().Update(ctx, cluster3)).Should(Succeed())
+
+			// it should create sliceconfig
+			Expect(k8sClient.Create(ctx, slice)).Should(Succeed())
+
+		})
+		AfterAll(func() {
+			// update sliceconfig tor remove clusters
+			createdSliceConfig := v1alpha1.SliceConfig{}
+			getKey := types.NamespacedName{
+				Namespace: sliceNamespace,
+				Name:      sliceName,
+			}
+			Expect(k8sClient.Get(ctx, getKey, &createdSliceConfig)).Should(Succeed())
+			createdSliceConfig.Spec.Clusters = []string{}
+			Expect(k8sClient.Update(ctx, &createdSliceConfig)).Should(Succeed())
+			// wait till workersliceconfigs are deleted
+			workerSliceConfigList := workerv1alpha1.WorkerSliceConfigList{}
+			ls := map[string]string{
+				"original-slice-name": slice.Name,
+			}
+			listOpts := []client.ListOption{
+				client.MatchingLabels(ls),
+			}
+			Eventually(func() bool {
+				err := k8sClient.List(ctx, &workerSliceConfigList, listOpts...)
+				if err != nil {
+					return false
+				}
+				return len(workerSliceConfigList.Items) == 0
+			}, timeout, interval).Should(BeTrue())
+
+			// it should delete sliceconfig
+			Expect(k8sClient.Delete(ctx, slice)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, cluster1)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, cluster2)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, cluster3)).Should(Succeed())
+
+			Expect(k8sClient.Delete(ctx, project)).Should(Succeed())
+
+		})
+		It("Should not allow deleting vpnkeyrotation config, if slice is present and raise an event", func() {
```

Comment:
Can we add one case +ve case where delete slice config should delete the VPN config too